src/lib.rs

```diff
@@ -92,6 +92,55 @@ where
     Ok(())
 }
 
+/// Describes a UCS-2 character as UTF-8 bytes.
+pub enum Utf8Bytes {
+    /// The first 128 characters that fit into a single byte.
+    Single(u8),
+    /// A character that takes two bytes.
+    Double(u8, u8),
+    /// A character that takes three bytes.
+    Triple(u8, u8, u8),
+}
+
+/// Decode UCS-2 string to UTF-8 with a custom callback function.
+///
+/// `output` is a function which receives every decoded character.
+pub fn decode_with<F>(input: &[u16], mut output: F) -> Result<usize>
```

Comment:
Could you also add a small unit test for this function? Thanks!

Reply:
Added. Also got rid of that enum. The `&[u8]` feels a bit nicer API to use.